integration-test/src/main.rs

```diff
@@ -0,0 +1,173 @@
+extern crate core;
+
+mod constants;
+mod gc;
+mod mock_metrics_server;
+
+use crate::constants::{GC_SPECIFIC_COUNTERS, GENERIC_JVM_COUNTERS, GENERIC_JVM_GAUGES};
+use crate::gc::JavaGarbageCollector;
+use crate::mock_metrics_server::CollectedRequest;
+use clap::Parser;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::net::{Ipv4Addr, SocketAddrV4};
+use std::path::PathBuf;
+use std::process::Command;
+use std::time::Duration;
+
+#[derive(Parser, Debug)]
+struct Args {
+    agent_jar_path: PathBuf,
+    gc: JavaGarbageCollector,
+    port: u16,
+}
+
+fn main() {
+    let args = Args::parse();
+
+    let collect_duration = Duration::from_secs(30);
+    let simple_host_app_path = PathBuf::from("java-src/simple-host-app");
+
+    // Compile a simple Java app that serves as the host for the agent tests
+    println!("Compiling Java host application...");
+
+    let javac_exit_status = Command::new("javac")
+        .current_dir(&simple_host_app_path)
+        .args(&["-d", ".", "App.java"])
+        .spawn()
+        .expect("Couldn't spawn javac process to build test application!")
+        .wait()
+        .expect("Couldn't wait for javac process completion!");
```

Comment:
[`status()`](https://doc.rust-lang.org/std/process/struct.Command.html#method.status) can be used here, since it does the spawn/wait for you:

```suggestion
        .status()
        .expect("Couldn't execute javac process to build test application!");
```